gherkin/ruby/lib/gherkin/stream/parser_message_stream.rb

```diff
@@ -11,7 +11,7 @@ def initialize(paths, sources, options)
         @sources = sources
         @options = options
 
-        id_generator = id_generator_class.new
+        id_generator = options[:id_generator] || id_generator_class.new
```

Comment:
I agree it's better to pass an `id_generator` object to the constructor. With that ability, I think we should remove the `:predictable_ids` option.

Offering multiple ways to do the same thing makes the library harder to maintain and use.

Reply:
makes sense, and only the gherkin binary used for the tests could instantiate that. I'll change this :)